File: Java/src/main/java/org/usfirst/frc/team3966/robot/Robot.java
/*
    2017 L&N STEMpunks

    lnstempunks.org

Authors:
    Cade Brown <cade@cade.site>

Website:
    programming.lnstempunks.org (or ln-stempunks.github.io)

 */
package org.usfirst.frc.team3966.robot;

import edu.wpi.first.wpilibj.IterativeRobot;
import edu.wpi.first.wpilibj.command.Command;
import edu.wpi.first.wpilibj.command.Scheduler;

import org.usfirst.frc.team3966.robot.subsystems.Drive;
import org.usfirst.frc.team3966.robot.commands.TankDrive;
import org.usfirst.frc.team3966.robot.commands.MecDrive;
import org.usfirst.frc.team3966.robot.commands.StopDrive;
import org.usfirst.frc.team3966.robot.values.IDs;
import org.usfirst.frc.team3966.robot.hardware.PS4Controller;
import org.usfirst.frc.team3966.util.Logger;

/**
 * Do not rename this file. It is auto generated, and used to run the robot.
 *
 * The class declaration should look like this:
 *
 * public class Robot extends IterativeRobot {
 *
 */
public class Robot extends IterativeRobot {

    public static Logger logger;

    public static Drive drive;
    public static PS4Controller controller;

    Command autonomousCommand;
    Command teleopCommand;

    /**
     *
     * @return
     */
    public String getModuleName() {
        return "WOPR-JR";
    }

    public String getModuleVersion() {
        return "0.0.1-java";
    }

    public void robotInit() {
        logger = new Logger(getModuleName(), getModuleVersion());

        drive = new Drive(IDs.LF_motor, IDs.LB_motor, IDs.RF_motor, IDs.RB_motor);
        controller = new PS4Controller(IDs.controller);

        // instantiate the command used for the autonomous period
        // teleopCommand = new TankDrive();
        teleopCommand = new MecDrive();

        // stay still during autonomous
        autonomousCommand = new StopDrive();
    }

    /*
    
        These functions are called right before autonomous/teleop
    
     */
    public void disabledInit() {

    }

    public void autonomousInit() {
        logger.info("Autonomous mode");
        if (autonomousCommand != null) {
            autonomousCommand.start();
        }
    }

    public void teleopInit() {
        logger.info("Teleop mode");
        if (autonomousCommand != null) {
            autonomousCommand.cancel();
        }

        if (teleopCommand != null) {
            teleopCommand.start();
        }
    }

    /**
     * This function is called periodically during autonomous/teleop
     */
    public void autonomousPeriodic() {
        Scheduler.getInstance().run();
    }

    public void teleopPeriodic() {
        Scheduler.getInstance().run();
    }

    public void disabledPeriodic() {
        Scheduler.getInstance().run();
        drive.stop();
    }

}

File: Python/src/robot.py
#!/usr/bin/env python3


"""

Our 2017 robot! (in python)

To deploy, run 

python robot.py deploy

Or, to simulate:

python robot.py sim

If you are another team, delete .deploy_cfg

"""

import wpilib
from commandbased import CommandBasedRobot

import subsystems

from commands.tankdrive import TankDrive
from commands.mecdrive import MecDrive
from commands.autonomous import Autonomous


class Robot(CommandBasedRobot):

    def getModuleName(self):
        return "WOPR-JR"

    def getVersion(self):
        return "0.0.0-py"

    def robotInit(self):
        subsystems.init()

        #self.teleopCommand = TankDrive()
        self.teleopCommand = MecDrive()
        self.autonomousCommand = Autonomous()


    def autonomousInit(self):
        self.autonomousCommand.cancel()
        self.autonomousCommand.start()

    def teleopInit(self):
        self.autonomousCommand.cancel()
        self.teleopCommand.start()


if __name__ == '__main__':
    wpilib.run(Robot)
